lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -95,16 +95,32 @@ namespace Js
         JavascriptExceptionObject *exception = nullptr;
 
         PROBE_STACK(scriptContext, Constants::MinStackDefault + spillSize + argsSize);
+        Js::JavascriptExceptionOperators::TryCatchFrameAddrStack tryCatchFrameAddrStack(scriptContext, frame);
 
```

Comment:
Moved this out of the try, because the destructor executes before the catch.

Reply:
Don't you want the dtor to run before the user catch code however?  I think you want to use {} to make that happen.

---
In reply to: [146085699](https://github.com/Microsoft/ChakraCore/pull/4017#discussion_r146085699) [](ancestors = 146085699)

Reply:
We want the dtor to run after the catch. We don't want to pop off the top most try catch addr.

Reply:
I don't think it is right, but doesn't matter until we start inlining functions with try/catches.  When running the user catch, the try isn't "active" anymore.  If you get an exception during the user catch, you need to unwind the inlined frames until the **next** try block.  But for now, if the try block is in the same function, we can't have inlined frames in between the catch and that try block...
We should still do it the right way in case we do enable inlining of functions with EH.